test/test_irods_http_api.py

```diff
@@ -3339,6 +3343,246 @@ def test_report_operation(self):
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'remove',
+                'name': zone_name
+            })
+            self.logger.debug(r.content)
+
+    def test_modifying_permissions_of_zone_collection(self):
+        rodsadmin_headers = {'Authorization': 'Bearer ' + self.rodsadmin_bearer_token}
+        rodsuser_headers = {'Authorization': 'Bearer ' + self.rodsuser_bearer_token}
+
+        # Add a remote zone to the local zone.
+        zone_name = 'tempZone'
+        #zone_name = 'other_zone'
+        #r = requests.post(self.url_endpoint, headers=rodsadmin_headers, data={
+        #    'op': 'add',
+        #    'name': zone_name
+        #})
+        #self.logger.debug(r.content)
+        #self.assertEqual(r.status_code, 200)
+        #self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+        try:
+            # Show the rodsuser cannot list the contents of the zone collection.
+            r = requests.get(f'{self.url_base}/collections', headers=rodsuser_headers, params={
+                'op': 'list',
+                #'lpath': '/'
+                'lpath': f'/{zone_name}'
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 400)
+            self.assertEqual(r.json()['irods_response']['status_code'], irods_error_codes.NOT_A_COLLECTION)
+
+            # Give the rodsuser permission to read the contents of the zone collection.
+            r = requests.post(self.url_endpoint, headers=rodsadmin_headers, data={
+                'op': 'set_zone_collection_permission',
+                'name': zone_name,
+                'user': self.rodsuser_username,
+                'permission': 'read'
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+            # Show the rodsuser can now list the contents of the zone collection.
+            r = requests.get(f'{self.url_base}/collections', headers=rodsuser_headers, params={
+                'op': 'list',
+                #'lpath': '/'
+                'lpath': f'/{zone_name}'
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+        finally:
+            # Remove the rodsuser's permissions from the zone collection.
+            r = requests.post(self.url_endpoint, headers=rodsadmin_headers, data={
+                'op': 'set_zone_collection_permission',
+                'name': zone_name,
+                'user': self.rodsuser_username,
+                'permission': 'null'
+            })
+            # Remove the remote zone.
+            #r = requests.post(self.url_endpoint, headers=rodsadmin_headers, data={
+            #    'op': 'remove',
+            #    'name': zone_name
+            #})
```

Comment:
Don't forget to handle this.

Reply:
See https://github.com/irods/irods_client_http_api/pull/282#discussion_r1669318629.